Revert ".devcontainer: nixfmt-rfc-style -> nixfmt"

MattSturgeon opened the pull request:
Reverts NixOS/nixpkgs#430330

As per https://github.com/NixOS/nixpkgs/issues/425583 and https://github.com/NixOS/nixpkgs/pull/430330#issuecomment-3156334137, I believe this was bumped prematurely.

Unless the devcontainer is using the in-tree copy of nixpkgs, the CI-pinned revision, or some other known-to-be up-to-date revision... If we can guarantee the devcontainer always uses an up-to-date nixpkgs, then this can be closed.

If the devcontainer is using nixpkgs from (e.g.) the NIX_PATH, then we don't know whether `pkgs.nixfmt` is the rfc-style formatter or the classic version. That's because NIX_PATH may contain an old revision or a stable release.

MattSturgeon committed:
Revert ".devcontainer: nixfmt-rfc-style -> nixfmt"

jfly reviewed (approved):
I know *nothing* about how devcontainers are used here, but this seems fine to me.

MattSturgeon committed:
.devcontainer: add TODO for nixfmt-rfc-style alias

The alias should be dropped only once `pkgs.nixfmt` is unlikely to be
nixfmt-classic.

I.e. after nixfmt 1.0 is in a stable release, but before the
nixfmt-rfc-style alias starts warning.

See https://github.com/NixOS/nixpkgs/issues/425583

wolfgangwalther reviewed (approved):
I know nothing about .devcontainer either, but `nixfmt-rfc-style` seems like the safe choice as long as it still exists.

MattSturgeon commented:
> It seems like this can be changed whenever the `nixfmt` in the container at https://github.com/devcontainers/features/pkgs/container/features%2Fnix is the same as `nixfmt-rfc-style`.

If so, should I amend the `TODO` comment to reference that instead of "a stable release"?
